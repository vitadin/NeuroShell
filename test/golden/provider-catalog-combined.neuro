%% Combined options test
\set[_echo_command="true"]
%% Tests combining multiple options together
\provider-catalog[provider=openai,sort=name]
\provider-catalog[search=chat,sort=name]
\provider-catalog[provider=anthropic,search=claude]
\provider-catalog[provider=openrouter,sort=name,search=unified]