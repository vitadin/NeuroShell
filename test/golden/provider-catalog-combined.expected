Setting _echo_command = true
%%> "\\provider-catalog[provider=openai,sort=name]"
Provider Catalog (Openai) (1 providers):
  [OAC] OpenAI Chat Completions (openai)
    Base URL: https://api.openai.com/v1
    Endpoint: /chat/completions
    Client Type: openai
    Description: OpenAI's chat completion API for GPT models
%%> "\\provider-catalog[search=chat,sort=name]"
Provider Catalog - Search: 'chat' (5 providers):
  [ANC] Anthropic Claude Chat (openai-compatible)
    Provider: anthropic
    Base URL: https://api.anthropic.com/v1
    Endpoint: /messages
    Client Type: openai-compatible
    Headers: anthropic-version: 2023-06-01
    Description: Anthropic Claude chat completions API
  [GMC] Google Gemini Chat (gemini)
    Provider: gemini
    Base URL: https://generativelanguage.googleapis.com/v1beta
    Endpoint: /models/{model}:generateContent
    Client Type: gemini
    Headers: Content-Type: application/json, x-goog-api-key: {API_KEY}
    Description: Google Gemini generative AI chat completions API
  [MSC] Moonshot Chat Completions (openai-compatible)
    Provider: moonshot
    Base URL: https://api.moonshot.ai/v1
    Endpoint: /chat/completions
    Client Type: openai-compatible
    Description: Moonshot AI chat completions API
  [OAC] OpenAI Chat Completions (openai)
    Provider: openai
    Base URL: https://api.openai.com/v1
    Endpoint: /chat/completions
    Client Type: openai
    Description: OpenAI's chat completion API for GPT models
  [ORC] OpenRouter Chat Completions (openai-compatible)
    Provider: openrouter
    Base URL: https://openrouter.ai/api/v1
    Endpoint: /chat/completions
    Client Type: openai-compatible
    Headers: HTTP-Referer: https://github.com/vitadin/NeuroShell, X-Title: NeuroShell
    Description: OpenRouter unified API access to multiple LLM providers
%%> "\\provider-catalog[provider=anthropic,search=claude]"
Provider Catalog (Anthropic) - Search: 'claude' (1 providers):
  [ANC] Anthropic Claude Chat (openai-compatible)
    Base URL: https://api.anthropic.com/v1
    Endpoint: /messages
    Client Type: openai-compatible
    Headers: anthropic-version: 2023-06-01
    Description: Anthropic Claude chat completions API
%%> "\\provider-catalog[provider=openrouter,sort=name,search=unified]"
Provider Catalog (Openrouter) - Search: 'unified' (1 providers):
  [ORC] OpenRouter Chat Completions (openai-compatible)
    Base URL: https://openrouter.ai/api/v1
    Endpoint: /chat/completions
    Client Type: openai-compatible
    Headers: HTTP-Referer: https://github.com/vitadin/NeuroShell, X-Title: NeuroShell
    Description: OpenRouter unified API access to multiple LLM providers